### DISCUSSION TITLE
Docker run configurations? (e.g IPC)

### DISCUSSION THREAD

**Discussion by eichelkrauta** [batect/batect#774] [Q&A] [answered]
Hey! Trying to set up a project with Batect, but I'm running into some issues with a Cypress bug that they've given a workaround for: https://github.com/cypress-io/cypress/issues/350

They mention that I'll need to set the docker run configuration `--ipc=host` as a workaround. 

Is this supported with Batect?

**Comment by charleskorn** [comment]
Hey @eichelkrauta, setting `ipc` isn't supported but is definitely something I could look at adding, as would allowing configuring `shmsize` ([as discussed here](https://github.com/cypress-io/cypress/issues/350#issuecomment-353572782)). 

It sounds like the root cause of the problem is that the default shared memory size is very small (64 MB) and Chrome crashes when it exhausts that. So being able to increase that to something larger would be helpful. (Setting `--ipc=host` gives the container access to the host's shared memory, which should be way larger than 64 MB, which is why this fixes the problem as well.)

In the meantime, have you tried mounting `/dev/shm` from the host into the container? A couple of people in that thread have suggested that as a workaround as well.

**Comment by charleskorn** [comment, marked as answer]
I've just added the ability to set the size of `/dev/shm` with [0.65.0](https://github.com/batect/batect/releases/tag/0.65.0), and also updated [the Cypress sample project](https://github.com/batect/batect-sample-cypress) to use it. Give that a go and let me know if that solves the problem for you.

**Reply by eichelkrauta** [reply]
Yes that indeed was the better fix! Thanks for digging deeper into that and resolving it so quickly. Also I really enjoy batect's upgrade process :) 

It did clear up the issue - we do have a larger site / rendering process.

**Reply by charleskorn** [reply]
No worries at all, glad to hear it's working now :)
